src/sqlfluff/rules/L065.py

```diff
@@ -0,0 +1,137 @@
+        results: List[LintResult] = []
+
+        # If len(set_operator) == 0 this will essentially not run
+        for set_operator in set_operator_segments:
+            preceeding_code = (
+                expression.reversed().select(start_seg=set_operator).first(sp.is_code())
+            )
+            following_code = expression.select(start_seg=set_operator).first(
+                sp.is_code()
+            )
+            res = {
+                "before": expression.select(
+                    start_seg=preceeding_code.get(), stop_seg=set_operator
+                ),
+                "after": expression.select(
+                    start_seg=set_operator, stop_seg=following_code.get()
+                ),
+            }
+
+            newline_before_set_operator = res["before"].first(sp.is_type("newline"))
+            newline_after_set_operator = res["after"].first(sp.is_type("newline"))
+
+            # If there is a whitespace directly preceeding/following the set operator we
+            # are replacing it with a newline later.
+            preceeding_whitespace = res["before"].first(sp.is_type("whitespace")).get()
+            following_whitespace = res["after"].first(sp.is_type("whitespace")).get()
+
+            if newline_before_set_operator and newline_after_set_operator:
+                continue
+            elif not newline_before_set_operator and newline_after_set_operator:
+                results.append(
+                    LintResult(
+                        anchor=set_operator,
+                        description=(
+                            "Set operators should be surrounded by newlines. "
+                            f"Missing newline before set operator {set_operator.raw}."
+                        ),
+                        fixes=_generate_fixes(whitespace_segment=preceeding_whitespace),
+                    )
+                )
+            elif newline_before_set_operator and not newline_after_set_operator:
+                results.append(
+                    LintResult(
+                        anchor=set_operator,
+                        description=(
+                            "Set operators should be surrounded by newlines. "
+                            f"Missing newline after set operator {set_operator.raw}."
+                        ),
+                        fixes=_generate_fixes(whitespace_segment=following_whitespace),
+                    )
+                )
+            else:
+                results.append(
+                    LintResult(
+                        anchor=set_operator,
+                        description=(
+                            "Set operators should be surrounded by newlines. "
+                            "Missing newline before and after set operator "
+                            f"{set_operator.raw}."
+                        ),
+                        # FIXME: Not sure how to make mypy happy here...
```

Comment:
Is this still an issue or resolved now? If so can we remove this line?

```suggestion
```

Reply:
It was "resolved" by `# type: ignore` .. So, no, it's not really resolved. I think it would be better to fix this properly. But couldn't figure a way to do it.